EKF/covariance.cpp

```diff
@@ -729,11 +729,42 @@ void Ekf::fixCovarianceErrors()
 		zeroRows(P,13,15);
 		zeroCols(P,13,15);
 	} else {
-		// constrain variances
-		for (int i = 13; i <= 15; i++) {
-			P[i][i] = math::constrain(P[i][i], 0.0f, P_lim[4]);
+		// Find the maximum delta velocity bias state variance and request a covariance reset if any variance is below the safe minimum
+		const float minSafeStateVar = 1e-9f;
+		float maxStateVar = minSafeStateVar;
+		bool resetRequired = false;
+		for (uint8_t stateIndex=13; stateIndex<=15; stateIndex++) {
+		    if (P[stateIndex][stateIndex] > maxStateVar) {
+			maxStateVar = P[stateIndex][stateIndex];
+		    } else if (P[stateIndex][stateIndex] < minSafeStateVar) {
```

Comment:
Suppose that the first axis  (stateIndex 13) was 1e-5. This would set maxStateVar to 1e-5.  Suppose the next axis checked was 1e-8 (safe). We do not want to trigger a reset yet. Suppose the last axis checked was 1e-10 (unsafe). Now we want to trigger a reset.